Use of external caches break guix build

Hello,

I'm trying out the guix build using the instructions [here](https://github.com/monero-project/monero/blob/master/contrib/guix/README.md), and the use of SOURCES_PATH and/or BASE_CACHE environmental variables seem to be causing builds to fail.

Some example output trying to build a single host after everything in the /gnu/store has been built:

```
$ env SOURCES_PATH="/path/to/sources-dir" BASE_CACHE="/path/to/base-dir" HOSTS='x86_64-linux-gnu' ./contrib/guix/guix-build
Checking that we can connect to the guix-daemon...

Hint: If this hangs, you may want to try turning your guix-daemon off and on
      again.

make: Entering directory '/home/guixuser/build/monero/contrib/depends'
make[1]: Entering directory '/home/guixuser/build/monero/contrib/depends'
make[1]: Leaving directory '/home/guixuser/build/monero/contrib/depends'
make: Leaving directory '/home/guixuser/build/monero/contrib/depends'
INFO: Building 977dedce2cf7 for platform triple x86_64-linux-gnu:
      ...using commit timestamp: 1742992626
      ...running at most 6 jobs
      ...from worktree directory: '/home/guixuser/build/monero'
          ...bind-mounted in container to: '/monero'
      ...in build directory: '/home/guixuser/build/monero/guix/guix-build-977dedce2cf7/build/distsrc-977dedce2cf7-x86_64-linux-gnu'
          ...bind-mounted in container to: '/distsrc-base/build/distsrc-977dedce2cf7-x86_64-linux-gnu'
      ...outputting in: '/home/guixuser/build/monero/guix/guix-build-977dedce2cf7/output/x86_64-linux-gnu'
          ...bind-mounted in container to: '/outdir-base/x86_64-linux-gnu'
x86_64-linux-gnu
Required environment variables as seen inside the container:
    DIST_ARCHIVE_BASE: /outdir-base/dist-archive
    VERSION: 977dedce2cf7
    HOST: x86_64-linux-gnu
    COMMIT_TIMESTAMP: 1742992626
    JOBS: 6
    DISTSRC: /distsrc-base/build/distsrc-977dedce2cf7-x86_64-linux-gnu
    OUTDIR: /outdir-base/x86_64-linux-gnu
    LOGDIR: /logdir-base/x86_64-linux-gnu
    OPTIONS: 
make: Entering directory '/monero/contrib/depends'
Fetching protobuf-cpp-3.21.12.tar.gz from https://github.com/protocolbuffers/protobuf/releases/download/v21.12/
/gnu/store/3jhfhxdf6v5ms10x5zmnl166dh3yhbr1-bash-minimal-5.1.16/bin/sh: line 1: curl: command not found
Fetching protobuf-cpp-3.21.12.tar.gz from https://downloads.getmonero.org/depends-sources
/gnu/store/3jhfhxdf6v5ms10x5zmnl166dh3yhbr1-bash-minimal-5.1.16/bin/sh: line 1: curl: command not found
make: *** [funcs.mk:300: /home/guixuser/build/guix_cache/sources_path/download-stamps/.stamp_fetched-native_protobuf-protobuf-cpp-3.21.12.tar.gz.hash] Error 127
make: Leaving directory '/monero/contrib/depends'
```

From the log it seems like neither the SOURCES_CACHE nor BASE_CACHE environmental variables are being passed to the container. Thus, I'm assuming `curl` can't be found because the container doesn't know to look somewhere else.

Any advice would be appreciated.

--- a/contrib/guix/guix-build
+++ b/contrib/guix/guix-build
@@ -444,7 +444,6 @@ EOF
                                  --keep-failed \
                                  --fallback \
                                  --link-profile \
-                                 --user="user" \
                                  --root="$(profiledir_for_host "${HOST}")" \
                                  ${SUBSTITUTE_URLS:+--substitute-urls="$SUBSTITUTE_URLS"} \
                                  ${ADDITIONAL_GUIX_COMMON_FLAGS} ${ADDITIONAL_GUIX_ENVIRONMENT_FLAGS} \

